shiny/render/transformer/_transformer.py

```diff
@@ -294,6 +300,24 @@ async def _run(self) -> OT:
         )
         return ret
 
+    def _repr_htmltools_(self) -> str | None:
+        import json
+
+        import htmltools
+
+        if self.default_output is None:
+            return None
+        res = htmltools.TagList(self.default_output(self.__name__)).render()
+        # TODO: Include deps too
+        return json.dumps(dict(html=res["html"]))
```

Comment:
This looks similar to the functionality in https://github.com/posit-dev/py-htmltools/pull/61

Reply:
Hmmm, that's interesting. I think we're coming at this from two different ways because you're targeting Quarto and I'm targeting IPython. And I guess Shiny's `output_ui` counts as a third approach.

Have you considered using a `<html-dependency>` custom element? It hadn't occurred to me before, but maybe that could be a unifying abstraction?

Reply:
BTW, in a similar vein, I have been using a `<shiny-bind>` custom element to wrap "HTML that possibly contains Shiny inputs/outputs" instead of calling `Shiny.bindAll`/`Shiny.unbindAll`, and it has worked great so far.

The way I've done it, it only binds during `connectedCallback` and unbinds during `disconnectedCallback` so it still isn't great when code is arbitrarily mutating stuff.